### PR TITLE
Feature/file version

### PR DESCRIPTION
**Pull request by chennan47**
<b>Purpose</b>
Fix the problem the osfstorge only provide the 10 latest version of files to user and add the scroll bar to make interface more readable for all file storage addon. Solve issue https://github.com/CenterForOpenScience/osf.io/issues/2059

<b>Changes</b>
the old file versions:
screen shot 2015-03-30 at 5 26 00 pm

the new file versions:
screen shot 2015-03-30 at 5 26 37 pm

<b>Facts</b>
the current file addons on OSF:
1. S3: S3 supports versions only if you enable the version for the bucket
2. Box: Box's version function is only available for premium user 
3. Dataverse: Doesn't support versioning
4. Dropbox: Dropbox currently only return 10 versions since the controller is defaulted to be 10. @chrisseto is going to change the waterbulter to make dropbox return more versions for osf users.
5. Github: Github currently returns 30 latest versions.
6. GoogleDrive: Google drive returns all versions.
7. Osfstorage: After this patch, it will returns all the versions.
8. figshare: figshare doesn't support versions.

### CODE REVIEW

**Comment by lbanner** [comment]
@chennan47 -- I appreciate you adding in those File Version facts -- That's good to have, and probably should include in some kind of Files-FAQ .... guess I will pursue that! Meantime, FYI @nicipfeiffer and @caileyfitz re: QA.

**Comment by chrisseto** [comment]
Quick update on the waterbutler side, dropbox will now return up to 250 revisions. thanks to @chennan47 for finding out it only returned 10!

**Comment by nicipfeiffer** [comment]
@lbanner Agreed. I am wonder where the version IDs come from?  For some docs there are crazy, large numbers like: 
screen shot 2015-03-31 at 9 42 30 am

**Comment by lbanner** [comment]
All dependent on the manner in which those add-ons handle versions .... no consistency that's for sure!

**Comment by chennan47** [comment]
@sloria rdy for review

File: website/static/js/osfHelpers.js [hunk 1]
```diff
@@ -390,6 +390,35 @@ var htmlEscape = function(text) {
...
 };
 
+/**
+ * Resize table to match thead and tbody column
+ */
+
+var tableResize = function(selector, checker) {
+        // Change the selector if needed
+    var $table = $(selector),
+        $bodyCells = $table.find('tbody tr:first').children(),
```
Comment (sloria) [inline website/static/js/osfHelpers.js]: > Use one var declaration per variable.

https://github.com/airbnb/javascript#variables

File: website/static/js/osfHelpers.js [hunk 2]
```diff
@@ -390,6 +390,35 @@ var htmlEscape = function(text) {
...
+
+    // Adjust the width of thead cells when window resizes
+    $(window).resize(function() {
+        // Get the tbody columns width array
+        colWidth = $bodyCells.map(function() {
+            return $(this).width();
+        }).get();
+        // Set the width of thead columns
+        $table.find('thead tr').children().each(function(i, v) {
+            if(i ===0 && $(v).width()>colWidth[i]){
```
Comment (sloria) [inline website/static/js/osfHelpers.js]: Fix spacing.

File: website/static/js/osfHelpers.js [hunk 3]
```diff
@@ -390,6 +390,35 @@ var htmlEscape = function(text) {
     return $('<div/>').text(text).html();
 };
 
+/**
+ * Resize table to match thead and tbody column
+ */
+
+var tableResize = function(selector, checker) {
```
Comment (sloria) [inline website/static/js/osfHelpers.js]: Ping @caneruguz for a 2nd review of this. Are there alternative solutions to this problem?

File: website/templates/project/view_file.mako [hunk 4]
```diff
@@ -75,18 +75,21 @@
             </button>
           </span>
 
-
+##          <div >
```
Comment (sloria) [inline website/templates/project/view_file.mako]: Was this unmatched? If so, go ahead and remove it.

**Comment by sloria** [comment]
Pass finished. Looks good. Pending @caneruguz's review of the table resizing function, this is good to merge.

File: website/static/js/osfHelpers.js (hunk 3 above)
Comment (caneruguz) [inline website/static/js/osfHelpers.js]: We talked with Nan about this. I know this is not an elegant or efficient solution, especially with the resize listener but I don't think there are many non-js options. The problem is that only the table body gets scrolling in this case and browser scrollbar positioning differences makes it hard to plan for a specific scenario. 

The html alternative would be to separate the thead from table and and scroll an outer layer of the table so the tbody never actually gets a scrollbar and absolute position the new table head. This is also not as elegant but reduces js load.

File: website/static/js/osfHelpers.js (hunk 1 above)
Comment (chennan47) [inline website/static/js/osfHelpers.js]: fixed in 33a4d79

File: website/static/js/osfHelpers.js (hunk 2 above)
Comment (chennan47) [inline website/static/js/osfHelpers.js]: fixed in 33a4d79

**Pull request closed without merging**